src/System.Management.Automation/engine/CommandCompletion/CompletionHelpers.cs

```diff
@@ -160,10 +160,12 @@ internal static string QuoteCompletionText(string completionText, string quote)
 
             string quoteInUse = string.IsNullOrEmpty(quote) ? "'" : quote;
 
-            if (quoteInUse == "'")
+            completionText = quoteInUse switch
             {
-                completionText = CodeGeneration.EscapeSingleQuotedStringContent(completionText);
-            }
+                "'" => completionText.Replace("'", "''"),
+                "\"" => completionText.Replace("`", "``").Replace("$", "`$"),
+                _ => completionText
```

Comment:
I'm sorry if my reasoning confused you. I think we need to keep the current behavior in refactoring for now. 
Please revert the changes.